FIX: exception on renaming control scheme (ISX-1544)

ritamerkl opened the pull request:
### Description

This PR fixes the exception captured in this [ticket](https://jira.unity3d.com/browse/ISX-1544).

### Changes made

provided old name to compare control scheme with serialized properties

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

ritamerkl committed:
fixed exception on renaming control scheme

jamesmcgill reviewed (approved):
Functionally it looks correct but it's not really clear here what is going on or why.

I don't understand what `m_OldName` is doing. Is it the `original` name or `initial` name of the control scheme?
Why would it be empty during save that it needs patched up?

ritamerkl committed:
Merge branch 'develop' into ISX-1544-exception-on-renaming-control-scheme

ritamerkl committed:
added explanation comments

ritamerkl committed:
Merge branch 'develop' into ISX-1544-exception-on-renaming-control-scheme

ritamerkl commented:
added some explanatory comments @jamesmcgill

ritamerkl committed:
format fix

jamesmcgill commented:
Things to consider:
- maybe it could be called `OriginalName` as that is how the comment clarifies it?
- maybe you don't need to to have conditional logic `IsNullOrEmpty` and just always just find by `OriginalName`. It should always have a name at the start. (This would remove the need to empty the variable as that might be fragile).
- or maybe there is some way to avoid the new temp variable completely (ensure the renaming got saved so the lookup always works or storing the original name in the state somewhere). Not sure that's possible.

ritamerkl committed:
refactored and fixed canceling renaming of control schemes

ritamerkl commented:
I refactored a bit. I changed it to storing the newName to make it simpler to not overwrite the old name if save is not pressed. Also I moved the Close method to the ControlSchemesView as the only usages of it are there. 
Unfortunately I will have to empty the variable OnClose because there I see no other (uncomplicated, clean) way to restore the oldName.
I tried one other way to avoid using the variable: that was to compare control schemes by Id's instead of their names (as I think this is also cleaner), it turned out to be completely over-engineered for such a small problem with a lot of overhead. I also did not want to bloat up the state with various variables to keep it clear. 
If you have another concrete idea I am open for it :) @jamesmcgill

jamesmcgill commented on Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/ControlSchemeCommands.cs:
```diff
@@ -32,13 +32,14 @@ public static Command RemoveDeviceRequirement(int selectedDeviceIndex)
                 var controlSchemeName = state.selectedControlScheme.name;
 
                 var controlSchemesArray = state.serializedObject.FindProperty(nameof(InputActionAsset.m_ControlSchemes));
+                //use the old name to find the control scheme in the array
```
Unneeded now

jamesmcgill commented on Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/ControlSchemeCommands.cs:
```diff
@@ -72,6 +73,8 @@ public static Command SaveControlScheme(bool updateExisting = false)
                 state.serializedObject.ApplyModifiedProperties();
 
+                if (!string.IsNullOrEmpty(newName))
+                    return state.With(selectedControlScheme: new InputControlScheme(controlScheme));
```
Should `newName` be used here, if it was tested in the line above?

jamesmcgill reviewed (approved):
Looks better now I think.
Still don't like having to remember to set to "" on close, but if there is no alternative then so be it.

ritamerkl committed:
clean up before merge

ritamerkl committed:
Merge branch 'develop' into ISX-1544-exception-on-renaming-control-scheme
